geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/GatewayReceiverCommand.java

```diff
@@ -793,24 +784,12 @@ private static void writeBatchException(Message origMsg, List<BatchException70>
   }
 
   private static void writeFatalException(Message origMsg, Throwable exception,
-      ServerConnection servConn, int batchId) throws IOException {
+      ServerConnection servConn) throws IOException {
     Message errorMsg = servConn.getErrorResponseMessage();
     errorMsg.setMessageType(MessageType.EXCEPTION);
     errorMsg.setNumberOfParts(2);
     errorMsg.setTransactionId(origMsg.getTransactionId());
-
-    // For older gateway senders, we need to send back an exception
-    // they can deserialize.
-    if ((servConn.getClientVersion() == null
```

Comment:
what about the `null` case?

Reply:
As it turns out it can never be `null` from `ServerConnection.getClientVersion()`. The implementation would explode with exceptions if it were. Would you like me to annotate `@NotNull` through this path?

Reply:
less code is better code